### PR TITLE
chore: verify automatic approval with capacity management

### PR DESCRIPTION
**Pull request by MatousJobanek**
verifies automatic approval and capacity management when using `HostOperatorConfig` CR
paired with: https://github.com/codeready-toolchain/host-operator/pull/288
depends on: https://github.com/codeready-toolchain/toolchain-common/pull/131

### CODE REVIEW

File: test/e2e/usersignup_test.go [hunk 1]
```diff
@@ -31,43 +32,142 @@ func (s *userSignupIntegrationTest) TearDownTest() {
 	s.ctx.Cleanup()
 }
 
-func (s *userSignupIntegrationTest) TestUserSignupApproval() {
-	s.T().Run("no approval config - manual approval by default", func(t *testing.T) {
+func (s *userManagementTestSuite) TestAutomaticApproval() {
```
Comment (alexeykazakov) [inline test/e2e/usersignup_test.go:35]: Are we mixing two suites in this file? `userSignupIntegrationTest` and `userManagementTestSuite`

File: test/e2e/usersignup_test.go [hunk 2]
```diff
@@ -31,43 +32,142 @@ func (s *userSignupIntegrationTest) TearDownTest() {
...
+
+			// then
+			err := s.hostAwait.WaitUntilMasterUserRecordDeleted("manual2")
+			require.NoError(t, err)
+		})
+	})
+
+}
+
+func (s *userManagementTestSuite) TestCapacityManagementWithManualApproval() {
```
Comment (alexeykazakov) [inline test/e2e/usersignup_test.go:108]: And here too.

File: testsupport/user_setup.go [hunk 3]
```diff
@@ -89,10 +88,14 @@ func CreateAndApproveSignup(t *testing.T, hostAwait *wait.HostAwaitility, userna
 	return *userSignup
 }
 
-func NewUserSignup(t *testing.T, hostAwait *wait.HostAwaitility, memberAwait *wait.MemberAwaitility, username string, email string) *toolchainv1alpha1.UserSignup {
-	memberCluster, ok, err := hostAwait.GetToolchainCluster(cluster.Member, memberAwait.Namespace, wait.ReadyToolchainCluster)
-	require.NoError(t, err)
-	require.True(t, ok)
+func NewUserSignup(t *testing.T, hostAwait *wait.HostAwaitility, memberAwait *wait.MemberAwaitility, username string, email string, setTargetCluster bool) *toolchainv1alpha1.UserSignup {
```
Comment (alexeykazakov) [inline testsupport/user_setup.go:91]: I would add a comment regarding `setTargetCluster` meaning here too.

**Review by alexeykazakov — COMMENTED** [review]
Overall looks good. Just have a few minor comments/questions.

File: test/e2e/base_user_test.go [hunk 4]
```diff
@@ -25,55 +21,10 @@ type baseUserIntegrationTest struct {
...
-	err = s.hostAwait.Client.Get(context.TODO(), types.NamespacedName{
-		Namespace: s.hostAwait.Namespace,
-		Name:      "toolchain-saas-config"},
-		cm)
-	require.NoError(s.T(), err)
-	require.Equal(s.T(), policy, cm.Data["user-approval-policy"])
-}
-
-func (s *baseUserIntegrationTest) createAndCheckUserSignup(specApproved bool, username string, email string,
+func (s *baseUserIntegrationTest) createAndCheckUserSignup(specApproved bool, username string, email string, setTargetCluster bool,
```
[the remark this answers is not in the document]
Comment (MatousJobanek) [reply test/e2e/base_user_test.go:31]: add here 8d9cacd

File: testsupport/user_setup.go (hunk 3 above)
Comment (MatousJobanek) [reply testsupport/user_setup.go:96]: added here 00aeba8

File: test/e2e/usersignup_test.go (hunk 1 above)
Comment (MatousJobanek) [reply test/e2e/usersignup_test.go:35]: I don't think so. This file contains tests for:
* automatic approval
* manual approval
* target cluster selection
* required verification
* username transformation

there is another file `user_management_test.go` that verifies everything regarding the user management - disabling, banning, deactivating.

Comment (alexeykazakov) [reply test/e2e/usersignup_test.go:35]: I'm confused. You are adding `func (s *userManagementTestSuite) TestAutomaticApproval() ` to `usersignup_test.go`
Should it be `func (s *userSignupIntegrationTest) TestAutomaticApproval()` instead?

File: test/e2e/usersignup_test.go (hunk 2 above)
Comment (alexeykazakov) [reply test/e2e/usersignup_test.go:108]: Same here:
```suggestion
func (s *userSignupIntegrationTest) TestCapacityManagementWithManualApproval() {
```

**Review by alexeykazakov — APPROVED** [review]
Besides the two remaining questions regarding the suite names the PR looks good.

File: test/e2e/usersignup_test.go (hunk 1 above)
Comment (MatousJobanek) [reply test/e2e/usersignup_test.go:35]: stupid me :man_facepalming:  - this slipped out of my notice. This is result of my bad copy-paste driven development. :man_facepalming: 
Thanks a lot for this catch :+1:

File: test/e2e/usersignup_test.go (hunk 2 above)
Comment (MatousJobanek) [reply test/e2e/usersignup_test.go:108]: Thanks

**Pull request merged**